tests/src/Content/Text/HTMLTest.php

```diff
@@ -58,6 +58,10 @@ public function dataHTMLText()
 				'expectedBBCode' => '[audio]http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3[/audio]',
 				'html' => '<audio src="http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3" controls="controls"><a href="http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3">http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3</a></audio>',
 			],
+			'bug-8075-html-tags' => [
+				'expectedBBCode' => "<rant here> I don't understand tests",
+				'html' => "<rant here> I don't understand tests",
```

Comment:
```suggestion
				'html' => '&lt;rant here&gt; I don&quot;t understand tests',
```

Reply:
The text arrives without the entities. The input is okay. So the test is failing. We mustn't change the input to match the output, but we have to possibly change the test to do the right stuff.

Reply:
If the text arrives without the entities when it is expected to be HTML, then we have to assume they are regular tags and `strip_tags` is doing its job.

We can have an additional test case like this:
```php
			'bug-8075-html-tags-unescaped' => [
				'expectedBBCode' => "I don't understand tests",
				'html' => "<rant here> I don't understand tests",
			],
```

And both tests would have the correct expected output. We just can't assume literal angle brackets in incoming HTML messages aren't tags.

Reply:
Remember tests don't care about the context, so theoretically we have to cover every possible context and provide the expected output for each. In practice we only cover the corner cases we encounter when fixing bugs because we already test obvious cases during development.

Reply:
Yeah, I had taken a wrong turn during my thinking about this :grin: